.github/workflows/ci.yml

```diff
@@ -55,47 +68,43 @@ jobs:
   Test-build-with-deps-matrix:
     name: BUILD WITH DEPS - JDK${{ matrix.jdk }} - ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
+    needs: wildfly-build
     strategy:
       fail-fast: false
       matrix:
         jdk: [11, 17]
         os: [ubuntu-20.04, windows-latest]
     steps:
-    - uses: n1hility/cancel-previous-runs@v2
-      with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
       with:
         repository: wildfly/wildfly
         ref: ${{ github.base_ref }}
         path: wildfly
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
       with:
         repository: wildfly/boms
         ref: ${{ github.base_ref }}
         path: boms
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
       with:
         path: quickstarts
-    - name: Set up JDK ${{ matrix.java }}
-      uses: AdoptOpenJDK/install-jdk@v1
+    - uses: actions/download-artifact@v3
```

Comment:
What's the purpose of this artifact download?

Reply:
Ok nevermind I saw in the WFLY CI workflow sources this is related with the reusable workflow. I should note that this makes the job look more complex / less readable than the simple checkout and mvn build tho ;-)

Reply:
By the way, is it possible to start the WFLY built and downloaded this way?

Reply:
There is no WildFly distribution built here, it's the maven repository which contains the dependencies required for provisioning a SNAPSHOT of WildFly for testing.